src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -202,14 +202,101 @@
+                ),
+                Ref("Expression_C_Grammar"),
+            ),
+        ),
+        AnyNumberOf(
+            OneOf(
+                Sequence(
+                    Sequence(
+                        Ref("BinaryOperatorGrammar"),
+                        Ref.keyword("NOT", optional=True),
+                    ),
+                    Ref("Expression_C_Grammar"),
+                    Sequence(
+                        Ref.keyword("ESCAPE"),
+                        Ref("Expression_C_Grammar"),
+                        optional=True,
+                    ),
+                ),
+                Ref("PatternMatchExpressionSegment"),
+                Sequence(
+                    Ref.keyword("NOT", optional=True),
+                    "IN",
+                    Bracketed(
+                        OneOf(
+                            Delimited(
+                                Ref("Expression_A_Grammar"),
+                            ),
+                            Ref("SelectableGrammar"),
+                            ephemeral_name="InExpression",
+                        )
+                    ),
+                ),
+                Sequence(
+                    Ref.keyword("NOT", optional=True),
+                    "IN",
+                    Ref("FunctionSegment"),  # E.g. UNNEST()
+                ),
+                Sequence(
+                    "IS",
+                    Ref.keyword("NOT", optional=True),
+                    Ref("IsClauseGrammar"),
+                ),
+                Ref("IsNullGrammar"),
+                Ref("NotNullGrammar"),
+                Sequence(
+                    # e.g. NOT EXISTS, but other expressions could be met as
+                    # well by inverting the condition with the NOT operator
+                    "NOT",
+                    Ref("Expression_C_Grammar"),
+                ),
+                Sequence(
+                    Ref.keyword("NOT", optional=True),
+                    "BETWEEN",
+                    # In a between expression, we're restricted to arithmetic operations
+                    # because if we look for all binary operators then we would match
+                    # AND as both an operator and also as the delimiter within the
+                    # BETWEEN expression.
+                    Ref("Expression_C_Grammar"),
+                    AnyNumberOf(
+                        Sequence(
+                            Ref("ArithmeticBinaryOperatorGrammar"),
+                            Ref("Expression_C_Grammar"),
+                        )
+                    ),
+                    "AND",
+                    Ref("Expression_C_Grammar"),
+                    AnyNumberOf(
+                        Sequence(
+                            Ref("ArithmeticBinaryOperatorGrammar"),
+                            Ref("Expression_C_Grammar"),
+                        )
+                    ),
+                ),
+            )
+        ),
+    ),
```

Comment:
I'm not sure we need all these changes.

I get your test SQL to pass with just this override in postgres and removing all your other changes (not just this one) and adding an override for `LikeGrammar`:

```suggestion
    LikeGrammar=OneOf("LIKE", "ILIKE", Sequence("SIMILAR", "TO")),
```